packages/theme/src/cli/utilities/theme-environment/theme-polling.ts

```diff
@@ -121,22 +122,32 @@ async function syncChangedAssets(
   localFileSystem: ThemeFileSystem,
   assetsChangedOnRemote: Checksum[],
 ) {
-  await Promise.all(
-    assetsChangedOnRemote.map(async (file) => {
-      if (localFileSystem.files.get(file.key)?.checksum === file.checksum) {
-        return
-      }
-      const asset = await fetchThemeAsset(targetTheme.id, file.key, currentSession)
-      if (asset) {
-        await localFileSystem.write(asset)
-        outputInfo(
-          outputContent`• ${timestampDateFormat.format(new Date())} Synced ${outputToken.raw('»')} ${outputToken.gray(
-            `download ${asset.key} from remote theme`,
-          )}`,
-        )
-      }
-    }),
+  const filesToGet = assetsChangedOnRemote.filter(
+    (file) => localFileSystem.files.get(file.key)?.checksum !== file.checksum,
   )
+
+  const chunks = batchedRequests(filesToGet, MAX_GRAPHQL_THEME_FILES, async (chunk) => {
```

Comment:
They're all slightly different in how they output (or don't) progress. I'm not sure what a good API would be for these...any ideas?

Reply:
Maybe we can consolidate how the CLI sends output in theme-polling and theme-downloader? Those are the 2 that output content differently which makes this hard. I don't have as much context here, @karreiro or @jamesmengo would know more

If not, we should use similar variable names / code structure maybe add a comment, so its easily identifiable to the reader these functions do the same thing.

Reply:
I was thinking if we could pass in the "way the function sends output" as a block to the theme querying logic so it can be unique per caller (theme-polling, theme-downloader, theme-reconciliation) but I see`theme-downloader` doesn't call a output function, but appends the output string to the output and the rendering is done downstream, which isn't super compatible to this passing-in-a-block idea

Reply:
Let's tackle this as a follow up issue.